packages/aws-cdk-lib/aws-lambda-event-sources/README.md

```diff
@@ -513,6 +552,131 @@ myFunction.addEventSource(new ManagedKafkaEventSource({
 }));
 ```
 
+Here's how to use Schema Registry with a self-managed Kafka cluster:
+
+```ts
+import { Secret } from 'aws-cdk-lib/aws-secretsmanager';
+import { SelfManagedKafkaEventSource, ConfluentSchemaRegistry } from 'aws-cdk-lib/aws-lambda-event-sources';
+
+// The list of Kafka brokers
+const bootstrapServers = ['kafka-broker:9092'];
+
+// The Kafka topic you want to subscribe to
+const topic = 'some-cool-topic';
+
+// The secret that allows access to your Confluent Schema Registry
+const secret = new Secret(this, 'Secret', { secretName: 'ConfluentSchemaRegistry' });
+
+declare const myFunction: lambda.Function;
+myFunction.addEventSource(new SelfManagedKafkaEventSource({
+  bootstrapServers: bootstrapServers,
+  topic: topic,
+  startingPosition: lambda.StartingPosition.TRIM_HORIZON,
+  provisionedPollerConfig: {
+    minimumPollers: 1,
+    maximumPollers: 3,
+  },
+  schemaRegistryConfig: new ConfluentSchemaRegistry({
+    schemaRegistryUri: 'https://self-managed-registry.example.com',
+    eventRecordFormat: lambda.EventRecordFormat.JSON,
+    authenticationType: lambda.KafkaSchemaRegistryAccessConfigType.BASIC_AUTH,
+    secret: secret,
+    schemaValidationConfigs: [{ attribute: lambda.KafkaSchemaValidationAttribute.KEY }],
+  }),
+}));
+```
+
+### Complete Example with IAM Permissions
+
+For a more comprehensive example that includes IAM policy configuration, here's a full CDK stack for MSK with AWS Glue Schema Registry:
+
+```ts
+import {
+  ManagedKafkaEventSource,
+  GlueSchemaRegistry,
+} from 'aws-cdk-lib/aws-lambda-event-sources';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import { App, StackProps, Stack } from 'aws-cdk-lib';
+import { CfnRegistry } from 'aws-cdk-lib/aws-glue';
+import * as iam from 'aws-cdk-lib/aws-iam';
+
+export class MskGlueSchemaRegistryStack extends Stack {
```

Comment:
It is an usual pattern for readme examples. Consider if Stack and App are needed here and if they are adding anything to the examples. Typically that's only the case if you want to show interaction between multiple stacks

Reply:
However, if you decide that a full example is the right way to go about this, you can add a new fixture [into this directory](https://github.com/aws/aws-cdk/tree/c8d61c9929ef461ccafab2a6e90f11ce635c5750/packages/aws-cdk-lib/rosetta/aws_lambda_event_sources) that doesn't wrap the code inside a stack.

Then specify the fixture you want to use like this:
~~~
```ts fixture=some-fixture
new Stack();
```
~~~
See https://github.com/aws/jsii-rosetta?tab=readme-ov-file#fixtures

I'd still recommend that this new fixture contains as much of the setup as possible, e.g. like the imports of `Stack` and `App`.

Reply:
Finally you may use `nofixture`, but that's really a last resort.

~~~
```ts nofixture
new Stack();
```
~~~